DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/register_map_and_streaming_interfaces/src/register_map_functor_model.cpp

```diff
@@ -0,0 +1,116 @@
+#include <sycl/ext/intel/fpga_extensions.hpp>
+#include <sycl/ext/intel/prototype/interfaces.hpp>
+#include <sycl/sycl.hpp>
+
+#include "exception_handler.hpp"
+
+using ValueT = int;
+
+// offloaded computation
+ValueT SomethingComplicated(ValueT val) { return (ValueT)(val * (val + 1)); }
+
+/////////////////////////////////////////
+
+struct FunctorRegisterMapIP {
+  // Use the 'register_map' annotation on a kernel argument to specify it to be
+  // a register map kernel argument.
+  register_map ValueT *input;
+  // Without the annotations, kernel arguments will be inferred to be register
+  // map kernel arguments if the kernel invocation interface is register mapped,
+  // and vise-versa.
+  ValueT *output;
+  // A kernel with a register map invocation interface can also independently
+  // have streaming kernel arguments, when annotated by 'conduit'.
+  conduit size_t n;
+  register_map_interface void operator()() const {
+    for (int i = 0; i < n; i++) {
+      output[i] = SomethingComplicated(input[i]);
+    }
+  }
+};
+
+int main(int argc, char *argv[]) {
+#if defined(FPGA_EMULATOR)
```

Comment:
Please update to the newest selectors:
```
    // device selector
#if FPGA_SIMULATOR
    auto selector = sycl::ext::intel::fpga_simulator_selector_v;
#elif FPGA_HARDWARE
    auto selector = sycl::ext::intel::fpga_selector_v;
#else  // #if FPGA_EMULATOR
    auto selector = sycl::ext::intel::fpga_emulator_selector_v;
#endif
```

This makes me think that you should probably have a HW target in your cmake and your readme - specially since you mention in your README that with this flow we can run quartus to get accurate fmax estimates

Reply:
selectors updated - though there's a known sycl resource issue that is preventing this feature from working on any sycl resource versions that's using the new selectors. https://hsdes.intel.com/appstore/article/#/14018088958